analyzer/tests/functional/fixit/test_fixit.py

```diff
@@ -0,0 +1,236 @@
+#
+# -------------------------------------------------------------------------
+#
+#  Part of the CodeChecker project, under the Apache License v2.0 with
+#  LLVM Exceptions. See LICENSE for license information.
+#  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+#
+# -------------------------------------------------------------------------
+
+"""
+Test case for the CodeChecker fixit command's direct functionality.
+"""
+
+
+import json
+import os
+import shutil
+import subprocess
+import unittest
+
+from distutils.spawn import find_executable
+
+from libtest import env
+
+
+class TestFixit(unittest.TestCase):
+    _ccClient = None
+
+    def setUp(self):
+
+        # TEST_WORKSPACE is automatically set by test package __init__.py .
+        self.test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self.test_workspace)
+
+        # Get the CodeChecker cmd if needed for the tests.
+        self._codechecker_cmd = env.codechecker_cmd()
+        self.report_dir = os.path.join(self.test_workspace, "reports")
+        # Change working dir to testfile dir so CodeChecker can be run easily.
+        self.__old_pwd = os.getcwd()
+
+    def tearDown(self):
+        """Restore environment after tests have run."""
+        os.chdir(self.__old_pwd)
+        if os.path.isdir(self.report_dir):
+            shutil.rmtree(self.report_dir)
+
+    @unittest.skipIf(find_executable('clang-apply-replacements') is None,
+                     "clang-apply-replacements clang tool must be available "
+                     "in the environment.")
+    def test_fixit_list(self):
+        '''
+        Test that the fixits are generated by clang-tidy
+        '''
+        # GIVEN
+        build_json = os.path.join(self.test_workspace, "build_simple.json")
+        source_file_cpp = os.path.join(self.test_workspace, "main.cpp")
+
+        # Create a compilation database.
+        build_log = [{"directory": self.test_workspace,
+                      "command": "clang++ -c " + source_file_cpp,
+                      "file": source_file_cpp
+                      }]
+
+        with open(build_json, 'w',
+                  encoding="utf-8", errors="ignore") as outfile:
+            json.dump(build_log, outfile)
+
+        # Test file contents
+        simple_file_content = """
+#include <vector>
```

Comment:
I created a main.cpp source file with the following content:
```cpp
#include <vector>

int main()
{
  std::vector<int> v;
  if (v.size() == 0)
    v.push_back(42);
}
```

I analyzed it so the fixit yaml files are generated into my report directory. Then I changed my source file (put an extra white space before the main function:
```cpp
#include <vector>


int main()
{
  std::vector<int> v;
  if (v.size() == 0)
    v.push_back(42);
}
```
Then I applied the fixits with the CodeChecker fix ./report command. My source file looked like this:
```cpp
#include <vector>

autot main( -> int)
{
  std::vector<int> v;
  if v.empty()0 {)
    v.push_back(42)

        };
}
```
At the CodeChecker parse command we handle source file changes and we print an error/warning message to the output that source files are changed and you have to renanalyze your project. I think we should implement the same logic at here too. What do you think @gyorb?